### DISCUSSION TITLE
Cache.invalidate how it works

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#2161] [Q&A] [answered]
Hi, I am trying to run `cache.invalidate` on one item, that is related to other types in the app.

So I have this let's say, `Parent` type that has relation to the `Child` type. (query for parent is relay paginated).

So when I do delete `Child` and I want to run `cache.invalidate`, it does invalidate it in many places, but in the fetched paginated list where the `Parent` is using it, it is still there? 

Am I doing something wrong or is it an urql error or can anyone at least point me in the correct direction, how to remove that (without doing some weird array-in-array-in-array filtering) :)

**Comment by JoviDeCroock** [comment, marked as answer]
This isn't really derivable from your example but I'm going out on a limb here to say you probably know which `Parent` you want to invalidate so let's look at this example in specific:

```js
const updates = {
  Mutation: {
    unlinkChildFromParent: (result, args, cache) => {
      if (result.unlinkChildFromParent) {
        // this means we'll invalidate Parent:id.child
        cache.invalidate(cache.keyOfEntity({ __typename: 'Parent', id: args.id }), 'child')
        // this means we'll just invalidate Child:id from the cache, this means that when we
        // get the parent that used to link to this before it will go
        // Parent:id.child --> Child:id --> oh it's missing
        cache.invalidate(cache.keyOfEntity({ __typename: 'Child', id: args.id }))
      }
    }
  }
}
```

**Reply by Miloshinjo** [reply]
Sorry I can't post the code as it is not open source :) But thank you again for your speedy answer, helpful as always :)
